it/test/connectors/ITSAStatusUpdateConnectorISpec.scala

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import _root_.helpers.{ComponentSpecBase, WiremockHelper}
+import connectors.constants.ITSAStatusUpdateConnectorConstants._
+import connectors.itsastatus.ITSAStatusUpdateConnector
+import connectors.itsastatus.ITSAStatusUpdateConnectorModel.{ErrorItem, ITSAStatusUpdateResponseFailure, ITSAStatusUpdateResponseSuccess}
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.http.Status.{BAD_REQUEST, NO_CONTENT, OK}
+import play.api.libs.json.Json
+
+class ITSAStatusUpdateConnectorISpec extends AnyWordSpec with ComponentSpecBase {
+
+  lazy val connector: ITSAStatusUpdateConnector = app.injector.instanceOf[ITSAStatusUpdateConnector]
+
+  "ITSAStatusUpdateConnector" when {
+
+    ".makeITSAStatusUpdate()" when {
+
+      "request body" should {
+
+        "be built correctly for OptOutReason" in {
+
+          WiremockHelper.stubPut(s"/income-tax-view-change/itsa-status/update/$taxableEntityId", NO_CONTENT, correctOptOutRequestBody, "{}")
+
+          val result = connector.makeITSAStatusUpdate(taxYear2024, taxableEntityId, optOutUpdateReason).futureValue
```

Comment:
Is it possible to treat `makeITSAStatusUpdate(...)` as private and test via the `optOut(...)` and `optIn(...)` functions?